tools/autograd/derivatives.yaml

```diff
@@ -761,7 +762,7 @@
 
 - name: polygamma_(Tensor(a!) self, int n) -> Tensor(a!)
   self: grad * polygamma(n + 1, self)
-  result: auto_element_wise
+  result: self_t.mul_(polygamma(n + 1, original_self_p))
```

Comment:
There used to be some `conj` here? But I'm guessing that's okay because polygamma doesn't support complex
It also looks like we avoid doing mul by self_t out of place and then copy_
Are these optimizations that we could automate with auto_element_wise/auto_linear as well?

Reply:
We have a grand total of 10 functions that have formulas for the inplace version specifically. So I don't think we want to bother adding such a logic to these automation.

Reply:
Ah, good point
What was is the reasoning behind having functions have formulas specifically for inplace anyway?

Reply:
A few of them are because there is no out-of-place version.
A few of them is for perf to avoid input cloning by providing an alternative implementation that only makes use for the output.